rustup-init.ps1

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env powershell
+
+# This is just a little script that can be downloaded from the internet to
+# install rustup. It just does platform detection, downloads the installer and
+# runs it.
+
+#XXX: If you change anything here, please make the same changes in setup_mode.rs
+function Write-Usage {
+    Write-Host @'
+rustup-init 1.18.3 (435397f48 2019-05-22)
+The installer for rustup
+
+USAGE:
+    rustup-init.exe [FLAGS] [OPTIONS]
+
+FLAGS:
+    -v, --verbose           Enable verbose output
+    -y                      Disable confirmation prompt.
+        --no-modify-path    Don't configure the PATH environment variable
+    -h, --help              Prints help information
+    -V, --version           Prints version information
+
+OPTIONS:
+        --default-host <default-host>              Choose a default host triple
+        --default-toolchain <default-toolchain>    Choose a default toolchain to install
+'@
+}
+
+function main([AllowEmptyCollection()] [string[]]$Argv = @()) {
+    foreach ($arg in $Argv) {
+        if ("$arg" -eq "-h" -or "$arg" -eq "--help") {
+            Write-Usage
+            return
+        }
+    }
+
+    Install-Rustup $Argv
+}
+
+function Get-Architecture {
+    $platform = [System.Environment]::OSVersion.Platform
+    switch -Wildcard ($platform) {
+        # Linux and Darwin both report as Unix
+        "Unix" {
+            $ostype = uname -s
+            $cputype = uname -m
+            break
+        }
+        "Win*" {
+            $ostype = (Get-WmiObject Win32_OperatingSystem).Caption
+            $cputype = (Get-WmiObject Win32_OperatingSystem).OSArchitecture
+            break
+        }
+        default {
+            Write-Error "could not determine OS platform: $platform"
+        }
+    }
+
+    switch -Wildcard ($ostype) {
+        "Darwin" {
+            $ostype = "apple-darwin"
+            break
+        }
+        "Linux" {
+            $ostype = "unknown-linux-gnu"
+            break
+        }
+        "*Windows*" {
+            $ostype = "pc-windows-msvc"
+            break
+        }
+        default {
+            Write-Error "unrecognized OS type: $ostype"
+        }
+    }
+
+    switch ($cputype) {
```

Comment:
Much easier to just use integer width, which is crossplatform and doesn't require string interpolation.

```
$is64bit = ([System.IntPtr]::Size -eq 8)
$is32bit = ([System.IntPtr]::Size -eq 4)
```

ref - https://ridicurious.com/2018/10/17/4-ways-to-find-os-architecture-using-powershell-32-or-64-bit/